Makefile

```diff
@@ -44,21 +45,34 @@ LDFLAGS = -X main.version=$(BRANCH) -X main.build=$(BUILD)
 # Rules
 all: clean build
 
-dependencies: $(DEPENDENCIES) $(VENDOR_PATH)
+dependencies: $(DEPENDENCIES) $(VENDOR_PATH) $(NOVENDOR_PACKAGES)
 
 $(DEPENDENCIES):
 	$(GO_GET) $@/...
 
+$(NOVENDOR_PACKAGES):
+	$(GO_GET) $@/...
+
 $(VENDOR_PATH):
 	$(GLIDE) install
 
-test: dependencies
-	$(GO_TEST) $(shell $(GLIDE) novendor)
+docker-build:
+	$(DOCKER_BUILD) -f Dockerfile.build -t $(DOCKER_BUILD_IMAGE) .
+
+test: dependencies docker-build
+	$(DOCKER_RUN) -v $(GOPATH):/go $(DOCKER_BUILD_IMAGE) make test-internal
+
+test-internal:
+	export TEST_NETWORKING=1; \
+	$(GO_TEST) $(NOVENDOR_PACKAGES)
+
+test-coverage: dependencies docker-build
+	$(DOCKER_RUN) -v $(GOPATH):/go $(DOCKER_BUILD_IMAGE) make test-coverage-internal
 
-test-coverage: dependencies
+test-coverage-internal:
 	export TEST_NETWORKING=1; \
 	echo "" > $(COVERAGE_REPORT); \
-	for dir in `$(GO_CMD) list ./... | egrep -v '/(vendor|etc)/'`; do \
+	for dir in `$(GO_CMD) list ./... | egrep -v '/vendor/'`; do \
```

Comment:
Maybe this shell eval has already been calculated and stored in $NOVENDOR_PACKAGES?